pkg/kubelet/dockertools/manager.go

```diff
@@ -216,12 +218,30 @@ func NewDockerManager(
 		cpuCFSQuota:            cpuCFSQuota,
 	}
 	dm.runner = lifecycle.NewHandlerRunner(httpClient, dm, dm)
-	if serializeImagePulls {
-		dm.imagePuller = kubecontainer.NewSerializedImagePuller(kubecontainer.FilterEventRecorder(recorder), dm, imageBackOff)
+	dm.containerGC = NewContainerGC(client, containerLogsDir)
+
+	minimumRequiredDockerAPIVersionForParallelImagePulls := false
+	version, err := dm.Version()
+	if err == nil {
+		result, err := version.Compare(MinimumDockerAPIVersion121)
+		if err == nil && result >=0 {
+			minimumRequiredDockerAPIVersionForParallelImagePulls = true
+		}
+	}
+
+	if minimumRequiredDockerAPIVersionForParallelImagePulls == false {
+		if serializeImagePulls.Value() {
+			dm.imagePuller = kubecontainer.NewSerializedImagePuller(kubecontainer.FilterEventRecorder(recorder), dm, imageBackOff)
+		} else {
+			dm.imagePuller = kubecontainer.NewImagePuller(kubecontainer.FilterEventRecorder(recorder), dm, imageBackOff)
+		}
 	} else {
-		dm.imagePuller = kubecontainer.NewImagePuller(kubecontainer.FilterEventRecorder(recorder), dm, imageBackOff)
+		if serializeImagePulls.Provided() == false {
+			dm.imagePuller = kubecontainer.NewImagePuller(kubecontainer.FilterEventRecorder(recorder), dm, imageBackOff)
+		} else {
+			dm.imagePuller = kubecontainer.NewSerializedImagePuller(kubecontainer.FilterEventRecorder(recorder), dm, imageBackOff)
```

Comment:
shouldn't this fall back to the `Value()` logic?  I would imaging that the general flow would look like this

```
unsafe --> default is serialized, if explicitly set use set value
safe     --> default is parallel, if explicitly set use set value
```

whereas here seems to be

```
if unsafe --> use value of flag (flag defaults to serialized)
if safe --> use parallel by default, but if the flag is set, use serialized regardless of flag value
```